### DISCUSSION TITLE
Release 3.0.0 installation

### DISCUSSION THREAD

**Discussion by StarflyerAlien** [earlephilhower/arduino-pico#1252] [General]
Colour me stupid, but I seem to be missing something here. (Probably a few brain cells!)

I have a working installation of pico-arduino on Windows 11 from last year. (Probably 2.6.something). I have downloaded the v3.0.0 .zip and unpacked it, but I can't see where to put the files. I have a bunch of files from the previous installation under an ArduinoData folder, but the file/folder structure is wildy different from the zip file structure. I don't seem to relate the two somehow.

Like I said, being short of a few brain cells, what am I missing? What obvious piece of instructions am I not seeing?

**Comment by earlephilhower** [comment]
>  ...I have downloaded the v3.0.0 .zip ...

That's your problem.  The ZIP files from GitHub to do not include all the submodules meaning that they're not useful for actually running anything.

Please use the `git` method described in the `README.md` or the Arduino Boards Installer (described same place) to get a working copy with **all** the files needed.

**Comment by StarflyerAlien** [comment]
Ah... OK. I already used the Arduino Boards Installer. It threw me at first as it told me that "Pico (Picoprobe)" was not a valid board. Then I spotted  the new "Upload Method", set it to "Picoprobe" and it started working.
Are you saying that updating the boards with Arduino Board Manager has done everything required,  including SDK 1.5.0? If so, that is absolutely brilliant.

Your pico-arduino package has saved me enormous amounts of hassle as I cannot get the new official rPi pico windows installer to work correctly, or at all, at times.

Thank you.


-------------------
No electrons were harmed by the sending of this email.
However, large numbers of electrons were slightly disturbed.
Electronic counselling has been offered.
-------------------
________________________________

**Reply by earlephilhower** [reply]
Yeah, the Boards Manager will take care of everything, all files.  Good luck!

**Comment by StarflyerAlien** [comment]
Excellent. I have a question. Is the entire installation self-contained with the Arduino data area without any external dependencies?
The reason I ask is that I have tried for some time to get the CMake/vsCode development system working... with varying degrees of failure. I initially took the 'Tom's Hardware' approach with no success. I uninstalled and deleted all of that and last week tried the new Windows installer... no success. What I am asking is whether there is any cross-dependency that could be causing a problem? I suspect not, but I have to ask. :-)

-------------------
No electrons were harmed by the sending of this email.
However, large numbers of electrons were slightly disturbed.
Electronic counselling has been offered.
-------------------
________________________________

**Reply by earlephilhower** [reply]
It's all self contained under the Arduino dirs, nothing else needed.  The sources and a toolchain are downloaded and installed by the board manager.
